### PR TITLE
React Navigation V5 update

### PR DESCRIPTION
**Pull request by MuckT**
### Description

Updates react-navigation and all related dependencies to highest v5 versions.

### Other changes

N/A

### Tested

Tested on iOS simulators and Android Emulators

### How others should test

Run the wallet on iOS and Android looking for issues related to navigation or animation. 

### Related issues

N/A

### Backwards compatibility

Yes

### CODE REVIEW

**Comment by MuckT** [comment]
@jeanregisser for seeing if we should replace the `react-native-tab-view+2.15.2.patch` with an updated patch is there specific behavior we can look for confirm if it is still needed?

**Comment by jeanregisser** [comment]
@MuckT could you double check on a real device? If I remember correctly this issue was only present on real devices.

**Comment by MuckT** [comment]
@jeanregisser I can check on a real iOS device I have already checked on a Samsung Galaxy S21 G5 for Android.

File: src/exchange/WithdrawCeloReviewScreen.tsx [hunk 1]
```diff
@@ -103,7 +103,7 @@ WithdrawCeloReviewScreen.navigationOptions = ({
   const onEdit = () => {
     ValoraAnalytics.track(CeloExchangeEvents.celo_withdraw_edit)
     const canGoBack = navigation
-      .dangerouslyGetState()
+      .getState()
```
Comment (kathaypacific) [inline src/exchange/WithdrawCeloReviewScreen.tsx:106]: how interesting that this isn't dangerous anymore 😂

**Comment by ValoraQA** [comment]
Hey @MuckT we have verified the above PR while performing BVT's and here are the below links for the same

**Test Rail links:**
**Android 9 :** https://valoraapp.testrail.io/index.php?/runs/view/134&group_by=cases:section_id&group_order=asc
**Android 11:** https://valoraapp.testrail.io/index.php?/runs/view/136&group_by=cases:section_id&group_order=asc
**iOS 14:** https://valoraapp.testrail.io/index.php?/runs/view/135&group_by=cases:section_id&group_order=asc
**iOS 15 :** https://valoraapp.testrail.io/index.php?/runs/view/133&group_by=cases:section_id&group_order=asc

Devices : iPhone 13mini (15.1.1), iPhone 12(14.7.1), Redmi Note 5 Pro (9.0), Samsung Galaxy F41 (11.0)

We will also update when we complete the regression testing on the given build and update the status with links in the comment 
Thanks..!!

**Comment by ValoraQA** [comment]
Hey @MuckT we have verified the above PR while performing regression and here are the below links for the same

**Test Rail links:**
**Android 10:** https://valoraapp.testrail.io/index.php?/runs/view/143&group_by=cases:section_id&group_order=asc
**iOS 14:** https://valoraapp.testrail.io/index.php?/runs/view/142&group_by=cases:section_id&group_order=asc

**Devices :** iPhone 12(14.7.1), Google Pixel XL(10.0)

Thanks..!!

**Pull request merged**